### DISCUSSION TITLE
Runtime error (assert failed) on print tree

### DISCUSSION THREAD

**Discussion by basdp** [foonathan/lexy#94] [Q&A] [answered]
I get this weird error when parsing a toy language grammar:
```
output.s: /app/example.cpp:5116: void lexy::parse_tree<lexy::_br8>::builder::cancel_production(marker &&) [Reader = lexy::_br8, TokenKind = void, MemoryResource = void]: Assertion `_cur.prod' failed.
```
I have created a reproduction in the playground:
https://lexy.foonathan.net/playground/?id=o5G6631dW&mode=tree

It happens when you try to parse an `expr_ident`. Is this a bug in Lexy or did I do something wrong? 
Apologies for not cleaning up the grammar, it could have been briefer.

**Comment by basdp** [comment]
As always, when you're stuck for hours, 5 minutes after posting the question, you solve it... 
I removed the `: lexy::transparent_production` inheritance from `expr_literal`, which removed the error and allowed the parse_tree to run.

Still would like to know if it is a Lexy bug or if I just don't understand how to use `lexy::transparent_production`.

**Reply by foonathan** [reply]
Productions marked with `lexy::transparent_production` don't appear in the parse tree. This is definitely a bug, lexy should never fail with an assertion failure. (There are some cases where preconditions of grammar rules could be checked at compile-time, but it would be too complex to do so, so they're checked at runtime instead. You'll get a useful error messages then, however).

**Comment by foonathan** [comment, marked as answer]
The bug has now been fixed and your example works.
